qa/test-db-instance-migration/test-migration/src/test/java/org/camunda/bpm/qa/upgrade/scenarios7210/jobexecutor/ExclusiveOverProcessHierarchiesTest.java

```diff
@@ -0,0 +1,230 @@
+public class ExclusiveOverProcessHierarchiesTest {
+
+    @Rule
+    public final UpgradeTestRule engineRule = new UpgradeTestRule();
+
+    RuntimeService runtimeService;
+    ManagementService managementService;
+    HistoryService historyService;
+
+    JobExecutor jobExecutor;
+    CommandExecutor commandExecutor;
+
+    ProcessEngineConfigurationImpl engineConfig;
+    JobState jobState;
+    boolean isJobExecutorAcquireExclusiveOverProcessHierarchies;
+
+    @Before
+    public void init() {
+        var engine = engineRule.getProcessEngine();
+        var engineConfig = (ProcessEngineConfigurationImpl) engine.getProcessEngineConfiguration();
+
+        this.runtimeService = engine.getRuntimeService();
+        this.managementService = engine.getManagementService();
+        this.historyService = engine.getHistoryService();
+
+        this.jobExecutor = engineConfig.getJobExecutor();
+        this.commandExecutor = engineConfig.getCommandExecutorTxRequired();
+        this.engineConfig = engineConfig;
+
+        storeOriginalStateBeforeTest(engine);
+    }
+
+    @After
+    public void tearDown() {
+        restoreOriginalStateAfterTest();
+    }
+
+    @Test
+    @ScenarioUnderTest("createRootProcessInstancesWithHierarchies.1")
+    public void shouldPerformLegacyAcquisitionWhenMultipleHierarchiesExclusiveJobsIsDisabled() {
+        // given
+
+        // The default configuration where exclusiveOverProcessHierarchies flag should be disabled by default
+        assertThat(engineConfig.isJobExecutorAcquireExclusiveOverProcessHierarchies()).isFalse();
+
+        // a deployed root Process which delegates to a subProcess (cardinality 2). See ExclusiveOverProcessHierarchiesScenario
+        var rootProcesses = runtimeService.createProcessInstanceQuery()
+                .processDefinitionKey("rootProcess")
+                .processInstanceBusinessKey("withMultipleHierarchies")
+                .list();
+
+        assertThat(rootProcesses).hasSize(2);
+
+        // 4 jobs are created (2 for each root process due to cardinality)
+        var jobs = managementService.createJobQuery()
+                .processDefinitionKey("subProcess")
+                .list();
+
+        assertThat(jobs).hasSize(4);
+
+        // all 4 jobs of the root process instances should have null rootProcessInstanceId since it didn't exist in the schema
+        assertThat(jobs.get(0).getRootProcessInstanceId()).isNull();
+        assertThat(jobs.get(1).getRootProcessInstanceId()).isNull();
+        assertThat(jobs.get(2).getRootProcessInstanceId()).isNull();
+        assertThat(jobs.get(3).getRootProcessInstanceId()).isNull();
+
+        var piJobs = assertProcessInstanceJobs(4, "subProcess", null);
+
+        int allJobsSize = getAllJobsSize();
+
+        // when
+
+        // The result of the AcquireJobsCmd (Component responsible for the job acquisition)
+        AcquiredJobs result = commandExecutor.execute(new AcquireJobsCmd(jobExecutor, allJobsSize));
```

Comment:
❓  Could we avoid acquiring all jobs? Isn't it possible to acquire only the jobs related to the scenario? The whole logic around the job state can be avoided that way.

Reply:
Since the unit here is the acquisition logic which is part of the `AcquireJobsCmd` and it shouldn't be controlled, i'm not sure if its possible to make the command return a window of jobs that only the test is interested. I'm open to ideas though.